internal/backends/collection.go

```diff
@@ -224,6 +226,50 @@ func (cc *collectionContract) Stats(ctx context.Context, params *CollectionStats
 	return res, err
 }
 
+// ListIndexesParams represents the parameters of Collection.ListIndexes method.
+type ListIndexesParams struct{}
+
+// ListIndexesResult represents the results of Collection.ListIndexesResult method.
+type ListIndexesResult struct {
+	Indexes []IndexInfo
+}
+
+// IndexInfo represents information about a single index.
+type IndexInfo struct {
+	Name   string
+	Key    []IndexKeyPair
+	Unique bool
+}
+
+// IndexKeyPair consists of a field name and a sort order that are part of the index.
+type IndexKeyPair struct {
+	Field string
+	Order IndexOrder
+}
+
+// IndexOrder represents the sort order of the index.
+type IndexOrder int32 // int32 to match the wire protocol
+
+const (
+	// IndexOrderAsc represents ascending sort order.
+	IndexOrderAsc IndexOrder = 1
+
+	// IndexOrderDesc represents descending sort order.
+	IndexOrderDesc IndexOrder = -1
+)
+
+// ListIndexes returns information about indexes in the database.
+//
+// If database or collection does not exist it returns ErrorCodeDatabaseDoesNotExist.
+func (cc *collectionContract) ListIndexes(ctx context.Context, params *ListIndexesParams) (*ListIndexesResult, error) {
+	defer observability.FuncCall(ctx)()
+
+	res, err := cc.c.ListIndexes(ctx, params)
+	checkError(err, ErrorCodeDatabaseDoesNotExist)
```

Comment:
That should be `ErrorCodeCollectionDoesNotExist` to match `DropCollection` and `RenameCollection`